src/PhpSpec/Listener/StateListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Listener;
+
+use PhpSpec\Event\ExampleEvent;
+use PhpSpec\Event\SuiteEvent;
+use PhpSpec\Message\MessageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class StateListener implements EventSubscriberInterface {
```

Comment:
`StateListener` is very generic, you can be more specific about what state it's listening for. It's something like 'where we are in the lifecycle of the test suite'?